DEVELOPER_GUIDE.md

```diff
@@ -141,6 +141,27 @@ Note - it may take a couple minutes to generate all the necessary bundles. If th
 [success][@osd/optimizer] 28 bundles compiled successfully after 145.9 sec, watching for changes
 ```
 
+### Run OpenSearch using Docker (Recommended for Mac M1 and M2 users)
+
+Mac users are advised to set up OpenSearch and OpenSearch Dashboards using [Docker]((https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/docs/docker-dev/docker-dev-setup-manual.md)). Follow the instructions in the Docker Developer Guide to set up OpenSearch using Docker.
+
+### Run OpenSearch Dashboards for Mac Users
+
+1. After setting up OpenSearch using Docker, make sure that you run the docker and the OpenSearch server instance is up and running. 
+2. Inside docker shell run:
+```$ yarn opensearch snapshot``` 
+3. Open new terminal, inside folder ```Opensearch-Dashboards```, configure environment variables for OpenSearch Dashboards:
+```
+export SERVER_HOST="localhost"
+export OPENSEARCH_HOSTS="http://localhost:9200"
+```
+4. Start the OpenSearch Dashboards development server using the start:docker command:
+```
+yarn start:docker
+```
+
+After following the previous instructions, OpenSearch Dashboards should be running on ```http://127.0.0.1:5603/fux```. To access the OpenSearch Dashboards web interface, use the URL ```http://127.0.0.1:5603/fux``` in your browser instead of ```localhost:5603/fux```. Changing the URL to use ```127.0.0.1``` instead of ```localhost``` can help avoid potential issues.
+
```

Comment:
So the docker dev environment setup instructions are in  https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/docs/docker-dev/docker-dev-setup-manual.md. And according to them opensearch should already  be up and running. If that didnt work for you, can you open an issue outlining what you faced when you tried to do so?

What these instructions do is start opensearch using the `snapshot` command. This will cause issues if an instance of opensearch is already ready. That seems wrong to me. We can use either `snapshot` or the opensearch instance from docker, but not both. Outlining what the issue you faced was and why this worked for you will make it easier for me to recommend how we can improve the instructions here!

Reply:
The Docker manual worked for me; however, I did not build the snapshot inside the Docker environment. Many MacBook M1 and M2 users have faced similar issues with the ```yarn opensearch snapshot``` command. Recently, some participants encountered the same problem, and following my instructions resolved their issues. You can see this problem discussed [here](https://dented-recess-c78.notion.site/Opensearch-Dashboards-TroubleShooting-Guide-bcb3632d18e24e5190aaa12f78edfd89).

Reply:
Seem like some people encounter similar issue when trying to follow the docker manual. I am not clear the reason behind this tho since if we are able to start the opensearch instance from docker, then we should already have a running opensearch instance without the need to use snapshot command again. @kavilla 

But thank you for your troubleshooting guide! I think instead of making changes to the developer guide, should we add these changes to the docker manual itself as i think that might be a better place? For example, maybe add a trouble shooting section underneath the docker manual? @ashwin-pc @joshuarrrr @Abyl10